src/main/kotlin/org/jetbrains/research/ml/dataset/anonymizer/transformation/util/anonymization/BaseAnonymizationVisitor.kt

```diff
@@ -0,0 +1,25 @@
+package org.jetbrains.research.ml.dataset.anonymizer.transformation.util.anonymization
+
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiRecursiveElementVisitor
+import org.jetbrains.research.ml.ast.transformations.anonymization.RenameUtil.renameElementDelayed
+
+open class BaseAnonymizationVisitor(file: PsiFile, private val anonymizer: BaseElementAnonymizer) :
+    PsiRecursiveElementVisitor() {
+    private val project = file.project
+
+    override fun visitElement(element: PsiElement) {
+        anonymizer.registerElement(element)
+        super.visitElement(element)
+    }
+
+    // TODO: it does not work for implements something now: see Java tests: classes_and_methods/out_2.java
```

Comment:
Please, rephrase it, ".. work for implements something now" doesn't match English grammar and barely conveys what you wanted to say